### PR TITLE
Add missing actions to data store documentation

### LINKED ISSUE #11337: Document missing `Actions`
While working on #10583, I noticed that the `Actions` of various data stores are not documented. This issue aims to document them.

### Cart Store
Location: `docs/third-party-developers/extensibility/data-store/cart.md`
- [ ] `setCartData`
- [ ] `setErrorData`
- [ ] `receiveCartContents`
- [ ] `receiveApplyingCoupon`
- [ ] `receiveRemovingCoupon`
- [ ] `receiveCartItem`
- [ ] `itemIsPendingQuantity`
- [ ] `itemIsPendingDelete`
- [ ] `setIsCartDataStale`
- [ ] `updatingCustomerData`
- [ ] `shippingRatesBeingSelected`
- [ ] `applyExtensionCartUpdate`
- [ ] `applyCoupon`
- [ ] `removeCoupon`
- [ ] `addItemToCart`
- [ ] `removeItemFromCart`
- [ ] `changeCartItemQuantity`
- [ ] `selectShippingRate`
- [ ] `setBillingAddress`
- [ ] `setShippingAddress`
- [ ] `updateCustomerData`

### Checkout
Location: `docs/third-party-developers/extensibility/data-store/checkout.md`
- [ ] `setPrefersCollection`

### Collections
Location: `docs/third-party-developers/extensibility/data-store/collections.md`
- [ ] `receiveCollectionError`
- [ ] `receiveLastModified`

### Validation
Location: `docs/third-party-developers/extensibility/data-store/validation.md`
- [ ] `clearAllValidationErrors`

### PR DESCRIPTION
**Pull request by tarunvijwani**
## What

Fixes #11337

## Why

The actions of various data stores were not documented.  With this PR, we're adding missing actions to data store documentation. 

## Testing Instructions

_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. Read the documentation and check for typos, grammar errors, etc.
2. Check all the links work.

* [x] Do not include in the Testing Notes 
* [x] Should be tested by the development team exclusively 

## Screenshots or screencast

N/A

## WooCommerce Visibility

Required:

* [ ] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental
* [x] N/A

## Checklist

Required:

* [x] This PR has either a `[type]` label or a `[skip-changelog]` label.
* [x] This PR is assigned to a milestone.

Conditional:

* [ ] This PR has a UI change and has been cross-browser tested at different viewport sizes on both the frontend and in the editor.
* [ ] This PR has a changelog description (if `[skip-changelog]` label is not present).
* [ ] This PR adds/removes a feature flag & I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR adds/removes an experimental interfaces, and I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR has been accessibility tested.
* [ ] This PR has had any necessary documentation added/updated.

## Changelog

> Skipping

### CODE REVIEW

File: docs/third-party-developers/extensibility/data-store/cart.md [hunk 1]
```diff
@@ -44,6 +67,365 @@ To utilize this store you will import the `CART_STORE_KEY` in any module referen
...
 
+## Actions
+
+### setCartData
+
+This action is used to set the cart data in the store.
+
+#### _Parameters_ <!-- omit in toc -->
+
+-   `cartData: Object`: The new cart data.
```
Comment (nielslange) [inline docs/third-party-developers/extensibility/data-store/cart.md]: For consistency-reasons, I suggest using the same formatting as we use in our other docs, e.g.
https://github.com/woocommerce/woocommerce-blocks/blob/6960ae656112b5a5249673db2aaa80e13c4b94c7/docs/third-party-developers/extensibility/data-store/collections.md?plain=1#L33-L40

What are your thoughts on documenting the inner structure of the `cartData` object, similar to what we're doing in https://github.com/woocommerce/woocommerce-blocks/blob/6960ae656112b5a5249673db2aaa80e13c4b94c7/docs/third-party-developers/extensibility/data-store/cart.md?plain=1#L188-L195

File: docs/third-party-developers/extensibility/data-store/cart.md [hunk 2]
```diff
@@ -44,6 +67,365 @@ To utilize this store you will import the `CART_STORE_KEY` in any module referen
...
+
+This action returns an action object used in updating the store with the provided cart. It omits the customer addresses so that only updates to cart items and totals are received.
+
+#### _Parameters_ <!-- omit in toc -->
+
+-   `cartContents: Object`: The new cart contents.
+
+#### _Returns_ <!-- omit in toc -->
+
+-   `Object`: The action object.
```
Comment (nielslange) [inline docs/third-party-developers/extensibility/data-store/cart.md]: Similar to the _Parameters_, what are your thoughts on documenting the inner structure of the objects within the _Returns_ sections? 

Here are two examples of other docs:

https://github.com/woocommerce/woocommerce-blocks/blob/6960ae656112b5a5249673db2aaa80e13c4b94c7/docs/third-party-developers/extensibility/data-store/collections.md?plain=1#L52

https://github.com/woocommerce/woocommerce-blocks/blob/6960ae656112b5a5249673db2aaa80e13c4b94c7/docs/third-party-developers/extensibility/data-store/collections.md?plain=1#L69-L74

File: docs/third-party-developers/extensibility/data-store/cart.md [hunk 3]
```diff
@@ -44,6 +67,365 @@ To utilize this store you will import the `CART_STORE_KEY` in any module referen
...
+```
+
+### itemIsPendingQuantity
+
+This action returns an action object to indicate if the specified cart item quantity is being updated.
+
+#### _Parameters_ <!-- omit in toc -->
+
+-   `cartItemKey: string`: The key of the cart item.
+-   `isPending: boolean`: The new status of whether the cart item quantity is being updated. Defaults to `true`.
```
Comment (nielslange) [inline docs/third-party-developers/extensibility/data-store/cart.md]: As mentioned earlier, for consistency-reasons, I'd use the same format we're using for our other docs. Here's an example of how other docs display default values and allowed values: https://github.com/woocommerce/woocommerce-blocks/blob/6960ae656112b5a5249673db2aaa80e13c4b94c7/docs/third-party-developers/extensibility/checkout-block/available-filters/coupons.md?plain=1#L149-L155

File: docs/third-party-developers/extensibility/data-store/cart.md [hunk 4]
```diff
@@ -44,6 +67,365 @@ To utilize this store you will import the `CART_STORE_KEY` in any module referen
...
+dispatch( setIsCartDataStale( isCartDataStale ) );
+```
+
+### updatingCustomerData
+
+This action returns an action object to indicate if the customer data is being updated.
+
+#### _Parameters_ <!-- omit in toc -->
+
+-   `isResolving: boolean`: The new status of whether the customer data is being updated.
```
Comment (nielslange) [inline docs/third-party-developers/extensibility/data-store/cart.md]: "The new status" sounds a bit strange. For consistency-reasons, I'd use the similar wording for all boolean parameters and returns. Here are two examples of other docs:

https://github.com/woocommerce/woocommerce-blocks/blob/6960ae656112b5a5249673db2aaa80e13c4b94c7/docs/third-party-developers/extensibility/checkout-block/available-filters/coupons.md?plain=1#L84

https://github.com/woocommerce/woocommerce-blocks/blob/6960ae656112b5a5249673db2aaa80e13c4b94c7/docs/third-party-developers/extensibility/checkout-block/available-filters/coupons.md?plain=1#L151

**Review by nielslange — COMMENTED** [review]
Thanks for working on this, @tarunvijwani. To make the docs easy accessible, I suggest that we use consistent formatting throughout our docs. I left a few notes with examples.

**Comment by tarunvijwani** [comment]
@nielslange Thank you so much for the suggestions! They were really helpful. I have updated the docs, please take a moment to review the PR again.

File: docs/third-party-developers/extensibility/data-store/checkout.md [hunk 5]
```diff
@@ -286,10 +288,27 @@ This is true when the total is being re-calculated for the order. There are nume
...
 
+## Actions
+
+### setPrefersCollection
+
+Sets the `prefersCollection` flag to true or false.
+
+#### _Parameters_ <!-- omit in toc -->
+
+-   `boolean`: True if the shopper prefers to collect their order.
```
Comment (nielslange) [inline docs/third-party-developers/extensibility/data-store/checkout.md]:
```suggestion
-   _prefersCollection_ `boolean`: True if the shopper prefers to collect their order.
```

**Review by nielslange — CHANGES_REQUESTED** [review]
Except for one missing parameter name, the PR looks great, @tarunvijwani.

**Comment by tarunvijwani** [comment]
Good catch, @nielslange. Thank you! I have applied your suggestions. 🎉

**Review by nielslange — APPROVED** [review]
Thanks for applying all changes. Let's ⛴️ this PR now, @tarunvijwani! 🙌

**Pull request merged**